### ISSUE TITLE
Inconsistent behaviour of Waterfall feature

### ISSUE THREAD

**Issue by SharmaAntriksh** [DaxStudio/DaxStudio#906]
I wrote a query to force the Batch event of SE and noticed that waterfall chart doesn't show up like 2 out of 5 times.

Code:

```
DEFINE
    MEASURE Sales[Big Sales] =
        CALCULATE ( COUNTROWS ( Customer ), Sales[Quantity] > 2, Sales )

EVALUATE
      SUMMARIZECOLUMNS (
          ROLLUPADDISSUBTOTAL ( Products[Color], "Color Total" ),
          "Big Sales", [Big Sales]
      )

```

![11](https://user-images.githubusercontent.com/73054674/187045084-2436b036-c7e6-4ccc-bcdb-60f699532682.png)

After a couple of re-runs:
![12](https://user-images.githubusercontent.com/73054674/187045168-a6dec7f6-c145-4eca-8328-c33ec8d53bbf.png)

**Comment by marcosqlbi** [comment]
I'm not surprised. The duration of the missing events is 0ms, probably the start and end times are the same, resulting in an "invisible" element. We could figure out a way to create "fictitious" bar, but to be honest I don't think it's important.
Indeed, in both examples you sent, the bar at line 6 is wrong, it cannot be after the batch it belongs to. It would be probably better to display either a smaller bar (because of the 0 length) or no bar at all to make the visualization more consistent with the actual data.

**Comment by dgosbell** [comment]
This appears to be due to rounding issues with these very short duration events. When you see this the start time for these 0 millisecond events is equal to the end time of the query so the bars are been drawn outside of the column width. I should be able to adjust the display to bring them back into view.

> Indeed, in both examples you sent, the bar at line 6 is wrong,

I think this is due to the fact that DAX Studio adjusts the duration of the batch events to subtract the child durations so that we show the "overhead" of the batch events to avoid double counting the child scan durations. It probably makes sense to show the true duration in the waterfall visual so that we can see the overlap

> It would be probably better to display either a smaller bar (because of the 0 length) 

There is actually code to display a smaller bar for 0's, but there is a bug that is scaling them just like the 1ms events, this will be fixed in the next release.

**Comment by dgosbell** [comment]
this should now be fixed in the 3.0.1 update
